server/writeModel/geocaching/cache.js

```diff
@@ -1,145 +1,200 @@
+  published: false,
+  countFindings: 0,
+  comments: [],
+  removed: false,
+
+  isAuthorized: {
+    commands: {
+      comment: { forAuthenticated: true, forPublic: false },
+      favor: { forAuthenticated: true, forPublic: false },
+      find: { forAuthenticated: true, forPublic: false },
+      hide: { forAuthenticated: false, forPublic: false },
+      publish: { forAuthenticated: true, forPublic: false },
+      remove: { forAuthenticated: false, forPublic: false }
+    },
+    events: {
+      commented: { forAuthenticated: true, forPublic: false },
+      favored: { forAuthenticated: true, forPublic: false },
+      found: { forAuthenticated: true, forPublic: false },
+      hidden: { forAuthenticated: false, forPublic: false },
+      published: { forAuthenticated: true, forPublic: false },
+      removed: { forAuthenticated: false, forPublic: false }
+    }
+  }
 };
 
 const commands = {
-	hide(cache, command, mark) {
-		cache.events.publish('hidden', {
-			name: command.data.name,
-			description: command.data.description,
-			coordinate: {
-				lat: command.data.coordinate.lat,
-				lon: command.data.coordinate.lon
-			}
-		});
-		mark.asDone();
-	},
-
-	publish(cache, command, mark) {
-		cache.events.publish('published', {
-			name: cache.state.name,
-			description: cache.state.description,
-			coordinate: {
-				lat: cache.state.coordinate.lat,
-				lon: cache.state.coordinate.lon
-			},
-      countFavorites: cache.state.countFavorites,
-    	countFindings: cache.state.countFindings,
-    	comments: cache.state.comments,
-    	removed: cache.state.removed,
-			published: true,
-		});
-		mark.asDone();
-	},
-
-	favor(cache, command, mark) {
-		cache.events.publish('favored', {
-			countFavorites: cache.state.countFavorites + 1
-		});
-		mark.asDone();
-	},
-
-	find(cache, command, mark) {
-		cache.events.publish('found', {
-			comments: [...cache.state.comments, {
-				text: command.data.text,
-				finder: command.user.id,
-				timestamp: command.metadata.timestamp
-			}],
-			countFindings: cache.state.countFindings + 1
-		});
-		mark.asDone();
-	},
-
-	comment(cache, command, mark) {
-		cache.events.publish('commented', {
-			comments: [...cache.state.comments, {
-				text: command.data.text,
-				author: command.user,
-				timestamp: command.metadata.timestamp
-			}],
-		});
-		mark.asDone();
-	},
-
-	remove(cache, command, mark) {
-		if(cache.state.published) {
-			 return mark.asRejected('An already published cache can not be removed');
-		}
-
-		cache.events.publish('removed');
-		mark.asDone();
-	}
+  comment: [
+    only.ifExists(),
+    only.ifValidatedBy({
+      type: 'object',
+      properties: {
+        text: { type: 'string', minLength: 1 }
+      },
+      required: [ 'text' ]
+    }),
+    onlyIfCacheHasBeenPublished(),
+    (cache, command, mark) => {
+      cache.events.publish('commented', {
+        comments: [ ...cache.state.comments, {
+          text: command.data.text,
+          author: command.user,
```

Comment:
Do you actually want to use the entire user object here, including the entire JWT token?

Reply:
I think the user id and some profile infos are enough. 👍